skyxplore-ui-test/src/test/java/com/mentoring/pageobject/FactoryPage.java

```diff
@@ -32,18 +45,35 @@ FactoryPage waitUntilPageLoads() {
         return this;
     }
 
-    public void clickExpansionItem() {
-        waitUntilClickable(expansionItem).click();
+    public FactoryPage selectConnectorExtender() {
+        waitUntilClickable(connectorExtenderButton).click();
+        return this;
     }
 
-    public FactoryPage clickCex01ManufacturingButton() {
-        waitUntilClickable(cex01ManufacturingButton).click();
+    public FactoryPage clickCex01ProductionButton() {
+        waitUntilClickable(cex01ProductionButton).click();
         return this;
     }
 
-    @Step("Asserting if expansion item under manufacturing is visible.")
-    public boolean isExpansionItemUnderManufacturingVisible() {
-        return isElementDisplayed(cex01ManufacturingLineItem);
+    @Step("Asserting if extender item in queue is visible.")
+    public boolean isExtenderItemInQueueVisible() {
+        return isElementDisplayed(cex01ProductionQueueItem);
+    }
+
+    @Step("Asserting quantity in production queue.")
+    public String getQuantityInQueue() {
+        return quantityInQueue.getText();
+    }
+
+    @Step("Check the production queue process started.")
+    public boolean isQueueProcessStarted() {
+        FluentWait wait = new FluentWait(driver)
+                .withTimeout(Duration.ofSeconds(15))
+                .pollingEvery(Duration.ofSeconds(1))
+                .ignoring(NoSuchElementException.class);
+        wait.until(ExpectedConditions.textToBePresentInElement(queueProcessBar, "00"));
```

Comment:
This seems to be a good reusable general method in Page.java. Also, since this returns a boolean value, it should be able to return both true and false. Currently it can only return ture, because the TimeoutException is not handled.

Reply:
Alternatively you can check out Awaitility instead of FluentWait. You can find an example in CharacterSelectionPage.java.

Reply:
Modified, thank you for your helping. :-)